cypress/e2e/login-flow.cy.js

```diff
@@ -0,0 +1,92 @@
+import { aliasQuery, aliasMutation, hasOperationName } from '../utils/graphql-test-utils'
+
+
+describe('empty spec', () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:3000/')
+  })
+
+  it('should have a title', () => {
+    cy.get('.title').should('have.text', 'Welcome to PupTrainer')
+  })
+
+  it('should tell the user to login', () => {
+    cy.get('.input-container > :nth-child(1)').should('have.text', 'Please login')
+  })
+
+  it('should have a username input field', () => {
+    cy.get('[placeholder="Username"]').should('have.attr', 'placeholder').should('eq', 'Username')
+  })
+
+  it('should have a email input field', () => {
+    cy.get('[placeholder="Email"]').should('have.attr', 'placeholder').should('eq', 'Email')
+  })
+
+  it('should have a register here link', () => {
+    cy.get(':nth-child(5) > a').should('have.text', 'Register here')
+  })
+
+  it('should take the user to a new url when clicking on the Register here link', () => {
+    cy.get(':nth-child(5) > a').should('have.text', 'Register here').click()
+    cy.url().should('eq', 'http://localhost:3000/create-user')
+  })
+
+  it('should display the create-user text fields', () => {
+    cy.visit('http://localhost:3000/create-user')
+    cy.get('.input-container > :nth-child(1)').should('have.text', 'Please enter a username and email')
+    cy.get('.input-container > :nth-child(5)').should('have.text', 'If already a user, click Here to login')
+  })
+
+  it('should create a user and be taken to the confirmation page', () => {
+    cy.intercept('POST', 'https://pup-trainer-api.herokuapp.com/graphql', (req) => {
+      const { body } = req
+      aliasMutation(req, 'createUser')
+      if (hasOperationName(req, 'createUser')) {
+        req.alis = 'gqlcreateUserMutation'
+        req.reply((res) => {
```

Comment:
Great job mocking the response from our queries!

Reply:
solid work on getting the stubs and intercepts working!